pkg/mimir/client/client.go

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/grafana/cortex-tools/blob/main/pkg/client/client.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Cortex Authors.
```

Comment:
Is this AGPL code? We're not going to be allowed to merge this if so; we're Apache 2.0-only.

Reply:
Gotcha, it's copied from Mimir's repo, so I'm sure we can work it out considering we own the license for both sides. I'll take this as an action item.

Reply:
It looks like it's copied from grafana/cortex-tools, which seems to be Apache 2 (but I might be missing something here)

Reply:
Yeah originally cortex Apache 2, copied into mimir and became AGPL, then copied here.

Reply:
As discussed in slack, I've solved this by rebasing onto the Apache 2 cortex client: https://github.com/grafana/agent/pull/2604/commits/e6feaad6a3399a0eb502155ec002b818643f5362, and adding a simple rename commit on top: https://github.com/grafana/agent/pull/2604/commits/74311ef9fc1ea30478fff94f5dbb81149b2e1241. All other commits were rebased from that point and are new code written for this hackathon. There is now a clear provenance back to cortex and all changes from there were done fresh for this PR. The end result is barely different, but it should be a valid use of the license now.